### PR TITLE
Update WorkManager sample to 2.7-rc

### PR DESCRIPTION
**Pull request by calren**
This PR contains the following:
- All commits from the `android-s` branch
- Fixes for merge conflicts when `android-s` branch was rebased with `main`
- Updates `SaveImageToGalleryWorker` to be a `CoroutineWorker` to fix the bug
```
java.lang.IllegalStateException: Expedited WorkRequests require a ListenableWorker to provide an implementation for `getForegroundInfoAsync()`
```
- Adds a NotificationUtils class that can be used by all Worker classes
- Updates from 2.7-beta01 to 2.7-rc01

### CODE REVIEW

File: WorkManagerSample/lib/src/main/java/com/example/background/workers/SaveImageToGalleryWorker.kt [hunk 1]
```diff
@@ -62,6 +62,14 @@ class SaveImageToGalleryWorker(appContext: Context, workerParams: WorkerParamete
         return Media.insertImage(
             resolver, bitmap, DATE_FORMATTER.format(Date()), TITLE
         )
+
+    }
+
+    override suspend fun getForegroundInfo(): ForegroundInfo {
+        // For a real world app you might want to use a different id for each Notification.
+        val notificationId = 2
```
[the remark this answers is not in the document]
Comment (calren) [reply WorkManagerSample/lib/src/main/java/com/example/background/workers/SaveImageToGalleryWorker.kt]: makes sense, done

File: WorkManagerSample/versions.gradle [hunk 2]
```diff
@@ -62,12 +62,12 @@ versions.rxjava2 = "2.1.3"
...
 versions.truth = "1.0.1"
-versions.work = "2.6.0"
+versions.work = "2.7.0-rc01"
 ext.versions = versions
 
 def build_versions = [:]
-build_versions.min_sdk = 14
-build_versions.compile_sdk = 29
+build_versions.min_sdk = 21
+build_versions.compile_sdk = 'android-S'
```
[the remark this answers is not in the document]
Comment (calren) [reply WorkManagerSample/versions.gradle]: done

File: WorkManagerSample/lib/src/main/java/com/example/background/workers/NotificationUtils.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+@TargetApi(Build.VERSION_CODES.O)
+fun createNotificationChannel(
+    context: Context,
+    channelId: String,
+    name: String
+): NotificationChannel {
+    val notificationManager =
+        context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+    return NotificationChannel(
+        channelId, name, NotificationManager.IMPORTANCE_LOW
```
Comment (keyboardsurfer) [inline WorkManagerSample/lib/src/main/java/com/example/background/workers/NotificationUtils.kt:68]: WDYT of making this a default parameter with `IMPORTANCE_LOW`?

Comment (calren) [reply WorkManagerSample/lib/src/main/java/com/example/background/workers/NotificationUtils.kt:68]: The notification still seems to show up when the app is running on a device with Android 11, but should we make this of higher importance?

**Review by tikurahul — APPROVED** [review]

**Pull request merged**